server/src/MessageHandler.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  TextDocumentPositionParams,
+  CancellationToken,
+  TextDocuments,
+  DocumentFormattingParams,
+  TextEdit,
+  Range,
+  Location,
+  DeclarationParams,
+  CompletionParams,
+  CompletionItem,
+  CompletionItemKind,
+} from 'vscode-languageserver'
+import * as util from './util'
+import { fullDocumentRange } from './provider'
+import { getDMMF } from '@prisma/sdk'
+import { TextDocument, Position } from 'vscode-languageserver-textdocument'
+import format from './format'
+
+export class MessageHandler {
+  constructor() {}
+
+  async handleHoverRequest(
+    params: TextDocumentPositionParams,
+    documents: TextDocuments<TextDocument>,
+  ) {
+    return null
+  }
+
+  getWordAtPosition(document: TextDocument, position: Position): String {
+    let currentLine = document.getText({
+      start: { line: position.line, character: 0 },
+      end: { line: position.line, character: 9999 },
+    })
+    // search for the word's beginning and end
+    var beginning = currentLine.slice(0, position.character + 1).search(/\S+$/)
+    var end = currentLine.slice(position.character).search(/\W/)
+    if (end < 0) {
+      return ''
+    }
+    return currentLine.slice(beginning, end + position.character)
+  }
+
+  async handleDefinitionRequest(
+    documents: TextDocuments<TextDocument>,
+    params: DeclarationParams,
+    _token?: CancellationToken,
+  ): Promise<Location> {
+    // TODO: Replace bad workaround as soon as ASTNode is available
+
+    const textDocument = params.textDocument
+    const position = params.position
+
+    const document = documents.get(textDocument.uri)
+
+    if (!document) {
+      return new Promise(resolve => resolve())
+    }
+
+    const documentText = document.getText()
+
+    let word = this.getWordAtPosition(document, position)
+    if (word == '') {
+      return new Promise(resolve => resolve())
+    }
+
+    // parse schem file to datamodel meta format (DMMF)
+    const dmmf = await getDMMF({ datamodel: documentText })
+
+    let modelName = dmmf.datamodel.models
+      .map(model => model.name)
+      ?.find(name => name == word)
+
+    // selected word is not a model type
+    if (modelName == null) {
+      return new Promise(resolve => resolve())
+    }
+
+    let modelDefinition = 'model '
+    // get start position of model type
+    let index = documentText.indexOf(modelDefinition + modelName)
+    const EOL = '\n'
```

Comment:
We should use `os.EOL` from https://nodejs.org/api/os.html#os_os_eol to avoid surprises

Reply:
Are we sure the file actually uses the OS EOL though? I can totally check out files with Linux EOL on my Windows machine, right?

Reply:
Yep, that is right 🙈 

We might be fine with `\n` in that case. os.EOL might break more things. Let's revert and maybe fix when someone reports if this is an actual problem.